### DISCUSSION TITLE
Updating a record in a basic collection using REST web API

### DISCUSSION THREAD

**Discussion by MikeCLX** [pocketbase/pocketbase#4996] [Q&A] [answered]
I have been searching all day but i cant seem to get past this issue, this is the first time posting so please be gentle :)

I am trying to update a record (patch) using the web API and VB.NET.
I can read the collection fine and authenticate fine but when i try to update i get a status code 400 "Bad Request"

CurrentAdminRecord.token is valid and it how i am reading the records
giftcards is a very basic collection with two custom fields void as boolean and tempbalance as number
I am trying to update the tempbalance to 90
i have replaced my domain with xxxxxxx in the posted code below
giftrecord.id contains a valid record id in giftcards

this is my code:
`Dim client = New HttpClient()
Dim keyValues1 = New List(Of KeyValuePair(Of String, String))()
client.DefaultRequestHeaders.Authorization = New AuthenticationHeaderValue("Bearer", CurrentAdminRecord.token)
keyValues1.Add(New KeyValuePair(Of String, String)("tempbalance", 90))

Dim MyContent As HttpContent = New FormUrlEncodedContent(keyValues1)

Dim response = Await client.PatchAsync(New Uri("https://xxxxxxxxx.com/api/collections/giftcards/records/" & giftrecord.id), MyContent)
`

what am i doing wrong?

**Comment by ganigeorgiev** [comment]
I'm not familiar with VB.Net but you could find more details about the 400 error in the _Admin UI > Logs_ section and/or start the executable with `--dev`.

**Reply by ganigeorgiev** [reply, marked as answer]
On second read, it seems that you are using `FormUrlEncodedContent` which I guess will send the request as `application/x-www-form-urlencoded` but we don't support this request content-type. The record operations works only with `application/json` and `multipart/form-data` requests.

_Based on the Microsoft docs you can try with [`JsonContent`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.json.jsoncontent?view=net-8.0) or [`MultipartFormDataContent`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.multipartformdatacontent?view=net-8.0)._

**Reply by MikeCLX** [reply]
Awesome thanks, I have it working now, I was thinking it was a JSON issue and this was the case.

Thanks for the push in the right direction
